backends/mongodb/mongodb.c

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#define BACKENDS_INTERNALS
+#include "mongodb.h"
+#include <mongoc.h>
+
+#define CONFIG_FILE_LINE_MAX ((CONFIG_MAX_NAME + CONFIG_MAX_VALUE + 1024) * 2)
+
+mongoc_client_t *mongodb_client;
+mongoc_collection_t *mongodb_collection;
+
+int mongodb_init(const char *uri_string, const char *database_string, const char *collection_string) {
+    mongoc_uri_t *uri;
+    bson_error_t error;
+
+    mongoc_init();
+
+    uri = mongoc_uri_new_with_error(uri_string, &error);
+    if(unlikely(!uri)) {
+       error("BACKEND: failed to parse URI: %s. Error message: %s", uri_string, error.message);
+       return 1;
+    }
+
+    mongoc_uri_set_option_as_int32(uri, MONGOC_URI_SOCKETTIMEOUTMS, 9000); // TODO: use variable for timeout
```

Comment:
This is a bool function, is it not a good idea at least to show an info message case it returns a false value?

Reply:
It can't return an error with this macros as a parameter, but I've added a condition with an error message for consistency.

Reply:
All right, OpenSSL has something like this too. Thanks!